snappi_ixnetwork/capture.py

```diff
@@ -155,57 +179,395 @@ def _config_missing_pallete(self, cap_filter, pallette, trigger, filter):
+                trigger,
+                filter,
+                21,
+                expressionString,
+            )
+        if not cap_filter.src.value == "00":
+            expressionString = self._config_common_filter_pallete(
+                cap_filter.src, pallette, trigger, filter, 22, expressionString
+            )
+        if not cap_filter.dst.value == "00":
+            expressionString = self._config_common_filter_pallete(
+                cap_filter.dst, pallette, trigger, filter, 38, expressionString
+            )
+        return expressionString
+
+    def _hex_to_str_with_space(self, hex_value):
+        return " ".join(
+            hex_value[i : i + 2] for i in range(0, len(hex_value), 2)
+        )
+
+    def _config_source_address(
+        self, src, pallette, trigger, filter, expressionString
+    ):
+        if src.value is not None:
+            pallette["SA1"] = self._hex_to_str_with_space(src.value)
+            if src.mask is not None:
+                pallette["SAMask1"] = self._hex_to_str_with_space(src.mask)
+            negate = ""
+            if src.negate is not None and src.negate is True:
                 filter["captureFilterSA"] = "notAddr1"
+                negate = "!"
             else:
                 filter["captureFilterSA"] = "addr1"
-            trigger["triggerFilterSA"] = filter["captureFilterSA"]
-        if cap_filter.dst.value is not None:
-            pallette["DA1"] = cap_filter.dst.value
-            if cap_filter.dst.mask is not None:
-                pallette["DAMask1"] = cap_filter.dst.mask
-            if (
-                cap_filter.dst.negate is not None
-                and cap_filter.dst.negate is True
-            ):
+            trigger["captureTriggerSA"] = filter["captureFilterSA"]
+            if len(expressionString) == 0:
+                expressionString = negate + "SA1"
+            else:
+                expressionString = expressionString + " and " + negate + "SA1"
+
+        return expressionString
+
+    def _config_destination_address(
+        self, dst, pallette, trigger, filter, expressionString
+    ):
+        if dst.value is not None:
+            pallette["DA1"] = self._hex_to_str_with_space(dst.value)
+            if dst.mask is not None:
+                pallette["DAMask1"] = self._hex_to_str_with_space(dst.mask)
+            negate = ""
+            if dst.negate is not None and dst.negate is True:
                 filter["captureFilterDA"] = "notAddr1"
+                negate = "!"
             else:
                 filter["captureFilterDA"] = "addr1"
-            trigger["triggerFilterDA"] = filter["captureFilterDA"]
+            trigger["captureTriggerDA"] = filter["captureFilterDA"]
+            if len(expressionString) == 0:
+                expressionString = negate + "DA1"
+            else:
+                expressionString = expressionString + " and " + negate + "DA1"
+        return expressionString
+
+    def _config_common_filter_pallete(
```

Comment:
This is not a must change.
calling multiple _config_common_filter_pallete is little annoying. is it not possible to maintain a class has all those property in a dics. Then loop through and call _config_common_filter_pallete once.